src/components/common/MetaProductDeletionBlock.tsx

```diff
@@ -0,0 +1,143 @@
+  metaProductType: MetaProductType
+  targetId: string
+  isDeleted: boolean
+}> = ({ metaProductType, targetId, isDeleted }) => {
+  const { formatMessage } = useIntl()
+  const [loading, setLoading] = useState(false)
+  const history = useHistory()
+  const { archiveActivity } = useArchiveMetaProduct()
+
+  let metaProduct = ''
+  switch (metaProductType) {
+    case 'Program':
+      metaProduct = formatMessage(messages.program)
+      break
+    case 'Activity':
+      metaProduct = formatMessage(messages.activity)
+      break
+    default:
+      break
+  }
+
+  const handleArchive = (metaProductType: MetaProductType, targetId: string) => {
+    setLoading(true)
+    switch (metaProductType) {
+      case 'Program':
+        break
+      case 'Activity':
+        archiveActivity({ variables: { activityId: targetId } })
+          .then(() => {
+            message.success(formatMessage(commonMessages.event.successfullyDeleted))
+            history.push(`/activities`)
+          })
+          .catch(handleError)
+          .finally(() => setLoading(false))
+        break
+      default:
+        break
+    }
+  }
+
+  if (!metaProductType || !targetId) return <Skeleton active />
+
+  return (
+    <AdminBlock>
+      <AdminBlockTitle>{formatMessage(messages.deleteProduct, { metaProduct })}</AdminBlockTitle>
+      <div className="d-flex align-items-center justify-content-between">
+        <div>
+          <div className="mb-2">{formatMessage(messages.deleteProductWarning, { metaProduct })}</div>
+          {metaProduct === 'Program' ? <StyledText>{formatMessage(messages.deleteProductDanger)}</StyledText> : null}
+        </div>
+
+        <AdminModal
+          title={formatMessage(messages.deleteProduct, { metaProduct })}
+          renderTrigger={({ setVisible }) =>
+            isDeleted ? (
+              <Button disabled>{formatMessage(commonMessages.status.deleted)}</Button>
+            ) : (
+              <Button type="primary" danger onClick={() => setVisible(true)}>
+                {formatMessage(messages.deleteProduct, { metaProduct })}
+              </Button>
+            )
+          }
+          okText={formatMessage(commonMessages.ui.delete)}
+          okButtonProps={{ danger: true, loading }}
+          cancelText={formatMessage(commonMessages.ui.back)}
+          onOk={() => handleArchive(metaProductType, targetId)}
+        >
+          <div>{formatMessage(messages.deleteConfirmation, { metaProduct })}</div>
+        </AdminModal>
+      </div>
+    </AdminBlock>
+  )
+}
+
+// program, activity, program_package, blog, merchandise
+const useArchiveMetaProduct = () => {
+  const [archiveActivity] = useMutation<hasura.ARCHIVE_ACTIVITY, hasura.ARCHIVE_ACTIVITYVariables>(gql`
+    mutation ARCHIVE_ACTIVITY($activityId: uuid) {
+      update_activity(where: { id: { _eq: $activityId } }, _set: { deleted_at: "now()" }) {
+        affected_rows
+      }
+    }
+  `)
+  return { archiveActivity }
+}
+
```

Comment:
NITS:You could think about how to arrange this hook when all of MetaProduct use this hook.
May like `getSimpleProduct` or other parts related to  `all products have the same action or logic` .